### PR TITLE
docs: keycloak idp

### PR DESCRIPTION
**Pull request by hifabienne**
### Definition of Ready

- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: docs/docs/guides/integrate/identity-providers/keycloak.mdx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import Activate from './_activate.mdx';
+import TestSetup from './_test_setup.mdx';
+
+<Intro provider="Keycloak"/>
+
+## Keycloak Configuration
+
+### Register a new client
+
+1. Login to your Keycloak Account and go to the clients list: <$KEYCLOAK-DOMAIN/auth/admin/$REALM/console/#/$REALM/clients>
```
Comment (mffap) [inline docs/docs/guides/integrate/identity-providers/keycloak.mdx:21]:
```suggestion
1. Login to your Keycloak account and go to the clients list: <$KEYCLOAK-DOMAIN/auth/admin/$REALM/console/#/$REALM/clients>
```

File: docs/docs/guides/integrate/identity-providers/keycloak.mdx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+2. Click on "Create Client"
+3. Choose OpenID Connect as Client Type and give your client an ID
+4. Enable Client authentication and the standard flow and direct access grants as authentication flow
+5. Add the Valid redirect URIs
```
Comment (mffap) [inline docs/docs/guides/integrate/identity-providers/keycloak.mdx:25]:
```suggestion
5. Add the valid redirect URIs
```

File: docs/docs/guides/integrate/identity-providers/keycloak.mdx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ - {your-domain}/ui/login/login/externalidp/callback
+ - Example redirect url for the domain `https://acme-gzoe4x.zitadel.cloud` would look like this:  `https://acme-gzoe4x.zitadel.cloud/ui/login/login/externalidp/callback`
+6. Go to the credentials tab and copy the secret
+
+![Add new OIDC Client in Keycloak](/img/guides/keycloak_add_client.png)
+![Get Client Secret](/img/guides/keycloak_client_secret.png)
+
+## ZITADEL Configuration
```
Comment (mffap) [inline docs/docs/guides/integrate/identity-providers/keycloak.mdx:33]:
```suggestion
## ZITADEL configuration
```

File: docs/docs/guides/integrate/identity-providers/keycloak.mdx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+### Add custom login policy
+
+<CustomLoginPolicy/>
+
+### Go to the IdP Providers Overview
```
Comment (mffap) [inline docs/docs/guides/integrate/identity-providers/keycloak.mdx:39]:
```suggestion
### Go to the IdP providers overview
```

File: docs/docs/guides/integrate/identity-providers/keycloak.mdx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+<IDPsOverview templates="Generic OIDC"/>
+
+### Create a new Generic OIDC Provider
```
Comment (mffap) [inline docs/docs/guides/integrate/identity-providers/keycloak.mdx:43]:
```suggestion
### Create a new generic OIDC provider
```

File: docs/docs/guides/integrate/identity-providers/keycloak.mdx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+<GenericOIDC
+    name=": e.g. Keycloak"
+    issuer=": The domain of your Keycloak account and /auth/realms/$REALM, Example: https://lemur-0.cloud-iam.com/auth/realms/acme"
```
Comment (mffap) [inline docs/docs/guides/integrate/identity-providers/keycloak.mdx:47]:
```suggestion
    issuer=": The domain where your Keycloak can be reached with the path /auth/realms/$REALM, Example: https://lemur-0.cloud-iam.com/auth/realms/acme"
```

**Review by mffap — CHANGES_REQUESTED** [review]

**Review by livio-a — APPROVED** [review]
as all requested changes are solved, i'll just approve to get this out 😃

**Pull request merged**